src/Database/Type/EnumType.php

```diff
@@ -111,16 +112,15 @@ public function toPHP(mixed $value, Driver $driver): ?BackedEnum
             return null;
         }
 
-        if (get_debug_type($value) !== $this->backingType) {
+        try {
+            return $this->enumClassName::tryFrom($value);
```

Comment:
This is needed to support PDO drivers that return strings.

`tryFrom()` will perform some coercion so string to integer is valid

Reply:
We could allow coercion only for string types if desired. Currently, it would allow a bool to be coerced.